quantum/process_keycode/process_caps_word.c

```diff
@@ -14,13 +14,59 @@
 
 #include "process_caps_word.h"
 
+#ifdef CAPS_WORD_INVERT_ON_SHIFT
+static uint8_t held_mods = 0;
+
+static bool handle_shift(uint16_t keycode, keyrecord_t* record) {
```

Comment:
No support for held oneshot shift (as it acts as a normal mod)?

Reply:
Thanks for the comment! I just added support for one-shot shifts. There is a limitation that under this option while Caps Word is on, one-shot shifts behave like regular shift keys, with effect only while the key is held. I updated the docs with a note about this behavior.